routes/panel.js

```diff
@@ -209,361 +214,389 @@ var freeChat = function(req, res) {
+        else { res.json("success"); }
+      });
+    });
+  }
 });
 
 router.post('/api/addShow', function(req, res) {
-	if (req.session.user == null) {
-		// not logged in, redirect to log in page
-		res.redirect('/panel');
-	}
-	else {
-		shows.addNewShow(req.body.title, req.body.day, req.body.time, [req.session.user.username], function(err, saved) {
-			if (err) {
-				console.log("failed to add show for user: ", err); 
-				res.json({"success": false, "err": err});
-			}
-
-			else {
-				// return full list of shows
-				res.redirect('/panel/api/usershows');
-			}
-		});
-	}
+  if (req.session.user == null) {
+    // not logged in, redirect to log in page
+    res.redirect('/panel');
+  }
+  else {
+    shows.addNewShow(req.body.title, req.body.day, req.body.time, [req.session.user.username], function(err, saved) {
+      if (err) {
+        console.log("failed to add show for user: ", err); 
+        res.json({"success": false, "err": err});
+      }
+
+      else {
+        // return full list of shows
+        res.redirect('/panel/api/usershows');
+      }
+    });
+  }
 });
 
 router.post('/api/showPic', function(req, res) {
-	if (req.session.user == null) {
-		// not logged in, redirect to log in page
-		res.redirect('/panel');
-	}
-	else {
-		shows.userHasAccessToShow(req.session.user.username, req.body.id, function(hasAccess) {
-			if (!hasAccess) { res.status(400).send(); }
-
-			else {
-				// user has access to update this show
-				var errorCallback = function(err) {
-					console.log("failed to add show picture: ", err);
-					res.status(400).send(err);
-				}
-
-				var picture = req.files.img.path.replace('public/', '/');
-				// update show data with new pictures
-				var newData = {"picture": picture};
-				shows.updateShow(req.body.id, newData, function(err, o) {
-					if (err) { errorCallback(err); }
-					else {
-						// updated successfully!
-						res.json("success");
-					}
-				});
-			}
-		});
-	}
+  if (req.session.user == null) {
+    // not logged in, redirect to log in page
+    res.redirect('/panel');
+  }
+  else {
+    shows.userHasAccessToShow(req.session.user.username, req.body.id, function(hasAccess) {
+      if (!hasAccess) { res.status(400).send(); }
+
+      else {
+        // user has access to update this show
+        var errorCallback = function(err) {
+          console.log("failed to add show picture: ", err);
+          res.status(400).send(err);
+        }
+
+        var imgPath = req.files.img.path;
+        var mimeType = req.files.img.mimetype;
+
+        // Perform image compression on JPG/PNG only
+        if (mimeType == "image/jpeg" || mimeType == "image/png") {
+          Jimp.read(imgPath, function(err, img) {
+            if (err) { errorCallback(err); }
+            img.resize(RESIZE_WIDTH, Jimp.AUTO).quality(IMAGE_QUALITY).write(imgPath);
+          });
+        }
+        else {
+          console.log("Non-png/jpg file type");
```

Comment:
Currently the `InputFileUpload` component has a property that lets you specify which file types will be accepted. We currently accept 'png/jpg/gif' which is why I added that check. However, the accepted file property can still be circumvented so I think in the future would should cause our API to return a 403 Forbidden.